eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/diagram/compositereference/SCADiagramUpdateCompositeReferenceFeature.java

```diff
@@ -172,6 +172,14 @@ public boolean update(IUpdateContext context) {
                 for (ComponentReference componentReference : component.getReference()) {
                     if (reference.getName().equals(componentReference.getName())) {
                         promotedReferences.add(componentReference);
+
+                        // fix the promote issue if we can here for SWITCHYARD-2836
+                        if (reference.getPromote().isEmpty()) {
+                            if (reference.getInterface().eClass().getName().
```

Comment:
In both cases, the Component and Composite references (unless they've been destroyed in a text editor) will have an interface, so yes - with about 90% certainty I'm going to say yes. The tooling won't allow anything else.